prelude/slang-cpp-types.h

```diff
@@ -226,21 +226,28 @@ struct Texture2D
 };
 
 /* Varying input for Compute */
-struct ComputeVaryingInput
+
+/* Used when running a single thread */
+struct ComputeThreadVaryingInput
 {
     uint3 groupID;
     uint3 groupThreadID;
 };
 
-struct GroupComputeVaryingInput
+struct ComputeVaryingInput
 {
     uint3 startGroupID;     ///< start groupID
     uint3 endGroupID;       ///< Non inclusive end groupID
 };
 
 /* Type that defines the uniform entry point params. The actual content of this type is dependent on the entry point parameters, and can be
-found via reflection or defined such that it matches the shader appropriately. */
+found via reflection or defined such that it matches the shader appropriately.
+*/
 struct UniformEntryPointParams;
+struct UniformState;
```

Comment:
Should the `UniformEntryPointParams` and `UniformState` type names be aligned better to indicate that they represent the "same" thing? E.g., `GlobalUniformParams` and `EntryPointUniformParams` would be my reccomendation.

A key point (which we'd need to actually test and make sure works...) is that two entry points compiled as part of the same "program" in Slang should be able to share the same `UniformState` (`GlobalUniformParams`), because our layout algorithm should guarantee that they agree on the layout of the global scope.

Reply:
I'm still not sold on forward-declaring the `UniformState` and `UniformEntryPointParams` types here as `struct` types, because it implies that all `ComputeFunc` callbacks will take the *same* type for these (it also encourages an application to define its own `UniformEntryPointParams` in a `.cpp` file and potentially violate the One Definition Rule.

I think it would be safer to have the actual callback types just take a `void*` (or maybe even `void const*`) for these uniform parameter blobs.